src/koans/14_recursion.clj

```diff
@@ -3,21 +3,27 @@
 
 (defn is-even? [n]
   (if (= n 0)
-    __
-    (___ (is-even? (dec n)))))
+    true
+    (not (is-even? (dec n)))))
 
 (defn is-even-bigint? [n]
   (loop [n   n
          acc true]
     (if (= n 0)
-      __
+      acc
       (recur (dec n) (not acc)))))
 
 (defn recursive-reverse [coll]
-  __)
+  (loop [coll coll  acc '()]
+    (if (= coll [])
```

Comment:
You can use more idiomatic ways to check if collection is empty with [empty](https://clojuredocs.org/clojure.core/empty_q) and [seq](https://clojuredocs.org/clojure.core/seq). For example: `(empty? %)` which returns boolean and `(seq %)` which return `nil` or sequable object.

Reply:
You can use a different approach and use [deconstruction](https://blog.brunobonacci.com/2014/11/16/clojure-complete-guide-to-destructuring/) (similar like Elixir) in the loop arguments.
```
(defn recursive-reverse [coll]
  (loop [[head & tail] coll
          new-coll    '()]
    (if-not head
      new-coll
      (recur tail (conj new-coll head)))))
```